### ISSUE TITLE
wxNotification generic implementation wrong window dimensions in GTK

### ISSUE THREAD

**Issue by bluehazzard** [wxWidgets/wxWidgets#24358]
#### Bug description:
Building wxWIdgets without libnotify results in an not usable wxNotification class on linux mint:
Creating a notification with the dialogs sample, creates a generic notification with a wrong window size.
As seen in the image below there is no "Message" text visible

![immagine](https://github.com/wxWidgets/wxWidgets/assets/5720389/87901d98-3ade-4e36-ba41-5aa8fb636958)

Stacking notifications work. Only the size of the dialog is wrong

### Platform and version information
- wxWidgets version  you use: 3.2.4
- wxWidgets port  you use: wxGTK
- OS  and its version: Linux Mint 21.3
  
  + GTK version: Gtk 3.24.33
  + Which GDK  backend is used: X.Org v: 1.21.1.4
  + Desktop environment Xfce 4.18.1
  + Current theme:

**Comment by vadz** [comment]
FWIW this works for me with Wayland and XWayland (i.e. explicitly using x11 GDK backend) under Gnome with Debian 12.

I guess the difference is in the WM behaviour, i.e. it's probably XFCE-specific.

**Comment by bluehazzard** [comment]
I have tried some things and got some results

```
diff --git a/src/generic/notifmsgg.cpp b/src/generic/notifmsgg.cpp
index 8566831470..5bbe3f539f 100644
--- a/src/generic/notifmsgg.cpp
+++ b/src/generic/notifmsgg.cpp
@@ -160,7 +160,7 @@ wxNotificationMessageWindow::wxNotificationMessageWindow(wxGenericNotificationMe
     m_messageTitle = new wxStaticText(m_messagePanel, wxID_ANY, wxString());
     m_messageTitle->SetFont(m_messageTitle->GetFont().MakeBold());
     textSizer->Add(m_messageTitle, wxSizerFlags(0).Border());
-    m_messageTitle->Hide();
+    //m_messageTitle->Hide();
     PrepareNotificationControl(m_messageTitle);

     m_messageText = new wxStaticText(m_messagePanel, wxID_ANY, wxString());
@@ -177,7 +177,7 @@ wxNotificationMessageWindow::wxNotificationMessageWindow(wxGenericNotificationMe

     wxSizer * const sizerTop = new wxBoxSizer(wxHORIZONTAL);
     sizerTop->Add(m_messagePanel, wxSizerFlags().Border(wxALL, FromDIP(1)));
-    SetSizer(sizerTop);
+    SetSizerAndFit(sizerTop);
 }

 wxNotificationMessageWindow::~wxNotificationMessageWindow()
```

![image](https://github.com/wxWidgets/wxWidgets/assets/5720389/363a699f-087e-4168-86f3-5481619c40ec)
Top notification is the first press on "Show" button 
Bottom notification is after a second press of "Show"

Any idea how i can debug this?

**Comment by bluehazzard** [comment]
i have tried on an arch linux distribution.
Compiling normally leads to a working notification by using the system notifications (uses libnotify by default).
Compiling with `./configure --with-libnotify=no` leads to the flawed notifications as described above. So the generic notification 

> FWIW this works for me with Wayland and XWayland (i.e. explicitly using x11 GDK backend) under Gnome with Debian 12.

are you sure you did not used libnotify?

**Comment by vadz** [comment]
No, I definitely didn't test with libnotify (it produces completely different notifications), but I did test using the "Use generic notifications" checkbox in this dialog which you don't have, presumably because you're building with `--without-libnotify`.
